server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletManagementIterator.java

```diff
@@ -279,6 +280,12 @@ private boolean shouldReturnDueToLocation(final TabletMetadata tm,
             || (tm.getHostingGoal() == TabletHostingGoal.ONDEMAND && tm.getHostingRequested()))) {
           return true;
         }
+        // If the Tablet has walogs and operation id then need to return so
+        // TGW can bring online to process the logs
+        if (tm.getLogs().isEmpty() && tm.getOperationId() != null
```

Comment:
```suggestion
        if (!tm.getLogs().isEmpty() && tm.getOperationId() != null
```

Reply:
@keith-turner - Oops, nice catch this is probably why the test i started to write didn't work as this logic was backwards. I can revisit the tests after you merge in your refactoring for #3904.